lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/StateMachineInitializer.java

```diff
@@ -33,4 +43,41 @@ public Map<String, State> initialize() throws IOException {
 
         return states;
     }
+
+    private File getConfigFile(String environment) {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        if (environment.contains("dev-")) {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/dev-statemachine-config.yaml"))
+                            .getFile());
+        }
+
+        if (environment.equals(BUILD_ENV)) {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/build-statemachine-config.yaml"))
+                            .getFile());
+        } else if (environment.equals(STAGING_ENV)) {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/staging-statemachine-config.yaml"))
+                            .getFile());
+        } else if (environment.equals(INTEGRATION_ENV)) {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/integration-statemachine-config.yaml"))
+                            .getFile());
+        } else {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/production-statemachine-config.yaml"))
+                            .getFile());
+        }
+    }
```

Comment:
Can we avoid all the duplicate file loading code by constructing the file name first? We could also make sure the env portion of the file name matches the env var, special case for dev excepted.